src/palace/manager/integration/metadata/lexile/api.py

```
@@ -0,0 +1,97 @@
+                        f"HTTP {response.status_code}. Check username and password.",
+                    )
+                raise IntegrationException(
+                    "Lexile API request failed",
+                    f"HTTP {response.status_code} for ISBN {isbn}",
+                )
+            return None
+
+        try:
+            data: dict[str, Any] = response.json()
+        except ValueError as e:
+            self.log.warning(f"Lexile API invalid JSON for ISBN {isbn}: {e}")
```

Ruff (0.15.7)

[warning] Logging statement uses f-string

(G004)